### PR TITLE
update fallback for backend id resolvers if stack, app id, or branch are in args

### PR DESCRIPTION
**Pull request by rtpascual**
## Problem
Came from https://github.com/aws-amplify/amplify-backend/pull/2028#discussion_r1768862801.

If stack, app id, or branch are passed in as an argument for generate commands, there is no intention to run these commands against a sandbox.

**Issue number, if available:**

## Changes

Only fallback to resolving sandbox backend identifier if stack, app id, and branch are not passed in as arguments to generate commands.

**Corresponding docs PR, if applicable:**

## Validation

Updated unit test.

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by sobolk — APPROVED** [review]

File: packages/cli/src/backend-identifier/backend_identifier_with_sandbox_fallback.ts [hunk 1]
```diff
@@ -18,23 +18,25 @@ export class BackendIdentifierResolverWithFallback
     private fallbackResolver: SandboxBackendIdResolver
   ) {}
   /**
-   * resolves the backend id, falling back to the sandbox id if there is an error
+   * resolves to deployed backend id, falling back to the sandbox id if there is an error and stack, appId, and branch are not provided
```
Comment (Amplifiyer) [inline packages/cli/src/backend-identifier/backend_identifier_with_sandbox_fallback.ts:21]: We are not falling back on errors

```suggestion
   * resolves to deployed backend id, falling back to the sandbox id if stack or appId and branch inputs are not provided
```

File: packages/cli/src/backend-identifier/backend_identifier_with_sandbox_fallback.ts [hunk 2]
```diff
@@ -18,23 +18,25 @@ export class BackendIdentifierResolverWithFallback
...
    */
   resolveDeployedBackendIdentifier = async (
     args: BackendIdentifierParameters
   ) => {
-    return (
-      (await this.defaultResolver.resolveDeployedBackendIdentifier(args)) ??
-      (await this.fallbackResolver.resolve())
-    );
+    if (args.stack || args.appId || args.branch) {
+      return await this.defaultResolver.resolveDeployedBackendIdentifier(args);
```
Comment (Amplifiyer) [inline packages/cli/src/backend-identifier/backend_identifier_with_sandbox_fallback.ts:27]: no need to await if we are not doing error handling.

Comment (rtpascual) [reply packages/cli/src/backend-identifier/backend_identifier_with_sandbox_fallback.ts:27]: We could possibly be throwing an error in the namespace resolver.

Comment (Amplifiyer) [reply packages/cli/src/backend-identifier/backend_identifier_with_sandbox_fallback.ts:27]: that will be caught by the caller when they `await` this method.

**Review by Amplifiyer — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Pull request merged**